src/player/Player.php

```diff
@@ -2248,6 +2280,15 @@ public function getSaveData() : CompoundTag{
 			$nbt->setInt(self::TAG_SPAWN_Z, $spawn->getFloorZ());
 		}
 
+		if($this->deathPosition !== null){
+			$nbt->setString(self::TAG_DEATH_WORLD, $this->deathPosition->getWorld()->getFolderName());
+			$nbt->setInt(self::TAG_DEATH_X, $this->deathPosition->getFloorX());
+			$nbt->setInt(self::TAG_DEATH_Y, $this->deathPosition->getFloorY());
+			$nbt->setInt(self::TAG_DEATH_Z, $this->deathPosition->getFloorZ());
+		}else{
+			$nbt->removeTag(self::TAG_DEATH_WORLD, self::TAG_DEATH_X, self::TAG_DEATH_Y, self::TAG_DEATH_Z);
```

Comment:
This is redundant. The NBT is freshly created, so it won't have these tags anyway.